google/cloud/bigquery/job.py

```diff
@@ -3082,10 +3085,25 @@ def done(self, retry=DEFAULT_RETRY, timeout=None):
             timeout (Optional[float]):
                 The number of seconds to wait for the underlying HTTP transport
                 before using ``retry``.
+            reload (Optional[bool]):
+                If ``True``, make an API call to refresh the job state of
+                unfinished jobs before checking. Default ``True``.
 
         Returns:
             bool: True if the job is complete, False otherwise.
         """
+        is_done = (
+            # Only consider a QueryJob complete when we know we have the final
+            # query results available.
+            self._query_results is not None
```

Comment:
checking: What about DDL statements, where we have no results?  Or would this already retain the empty row iterator?

Reply:
DDL statements don't 404 for `jobs.getQueryResults`, do they? This is the full response object of `jobs.getQueryResults`, not just the rows.